ui-app/client/src/components/TextBox/TextBox.tsx

```diff
@@ -154,8 +163,8 @@ function TextBox(props: ComponentProps) {
 				true,
 			);
 	}, [value, validation]);
-
-	const updateStoreImmediately = upStoreImm || autoComplete === 'on';
+	const changeEvent = onChange ? props.pageDefinition.eventFunctions[onChange] : undefined;
+	const updateStoreImmediately = upStoreImm || autoComplete === 'on' || changeEvent;
```

Comment:
I clearly told you not to do this. onChange event should respect not upStoreImm. Please remove this and add the required logic.